python-lib/dku_deeplearning_image/utils.py

```diff
@@ -77,25 +78,39 @@ def can_use_gpu():
     return len(tf.config.experimental.list_physical_devices('GPU')) > 0
 
 
-def set_gpu_options(should_use_gpu, gpu_list, memory_limit):
+def set_gpu_options(should_use_gpu, gpu_list, gpu_memory_allocation_mode, memory_limit_ratio=None):
     log_info("load_gpu_options")
     if should_use_gpu and can_use_gpu():
         log_info("should use GPU")
         gpus = tf.config.experimental.list_physical_devices('GPU')
+        for i, g in enumerate(gpus):
+            g.id = i
         gpus_to_use = [gpus[int(i)] for i in gpu_list] or gpus
-        if memory_limit:
+        log_info("GPUs on the machine: {}".format(str([g.id for g in GPUtil.getGPUs()])))
+        log_info("Will use the following GPUs: {}".format(gpus_to_use))
+        if gpu_memory_allocation_mode == constants.GPU_MEMORY_LIMIT and memory_limit_ratio:
             for gpu in gpus_to_use:
+                memory_limit = calculate_gpu_memory_allocation(memory_limit_ratio, gpu)
+                log_info("Restraining GPU(s) {} to {} Mo".format(gpu, str(memory_limit)))
```

Comment:
What if GPUs have different memory sizes? Perhaps better to log only the ratio?

Reply:
Also, no need to `str` and proposition to use f"" string formatting

Reply:
Prefere to do both to see how conversions worked. As I display this for each GPU, there are no problem if each GPU has a different memory size.